src/jit/x64/emit.dasc

```diff
@@ -1520,6 +1600,28 @@ void MVM_jit_emit_primitive(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJ
         | mov aword WORK[dst], TMP6;
         break;
     }
+    case MVM_OP_objprimunsigned: {
+        MVMint16 dst  = ins->operands[0].reg.orig;
+        MVMint16 type = ins->operands[1].reg.orig;
+        | mov TMP5, aword WORK[type];
+        | test TMP5, TMP5;
+        | jz >1;
+        | mov ARG1, TC;
+        | mov ARG2, OBJECT:TMP5->st;
+        | mov FUNCTION, STABLE:ARG2->REPR;
+        | mov FUNCTION, REPR:FUNCTION->get_storage_spec;
+        | call FUNCTION;
+        | mov TMP6, STORAGESPEC:RV->boxed_primitive;
+        | cmp TMP6, 1;
```

Comment:
`test TMP6, TMP6` is shorter and simpler assembly

Reply:
Matter of fact, you might be able to do:

     mov TMP6, STORAGESPEC:RV->boxed_primitive;
     and TMP6, STORAGESPEC:RV->is_unsigned;
     mov WORK[dst], TMP6;

Provided both are actually 1 when you need them (which you rely on now anyway)

Reply:
I'm not sure that'd work, `TMP6` can be the value of any of the `MVM_STORAGE_SPEC_BP` constants. src/core/interp.h specifically compares it against 1

Reply:
Hmmm. Mabe not then (although we could always do a final `AND TMP6, 1`). It's not important :smile: